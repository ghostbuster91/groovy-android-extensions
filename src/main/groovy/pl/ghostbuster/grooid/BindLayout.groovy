package pl.ghostbuster.grooid

import org.codehaus.groovy.transform.GroovyASTTransformationClass

import java.lang.annotation.ElementType
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import java.lang.annotation.Target

@Retention(RetentionPolicy.SOURCE)
@Target([ElementType.TYPE])
@GroovyASTTransformationClass(classes = [BindLayoutASTTransformation])
@interface BindLayout {
}
