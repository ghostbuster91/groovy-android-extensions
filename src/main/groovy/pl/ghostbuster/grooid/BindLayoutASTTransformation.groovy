package pl.ghostbuster.grooid

import org.codehaus.groovy.ast.ASTNode
import org.codehaus.groovy.control.CompilePhase
import org.codehaus.groovy.control.SourceUnit
import org.codehaus.groovy.transform.AbstractASTTransformation
import org.codehaus.groovy.transform.GroovyASTTransformation

@GroovyASTTransformation(phase = CompilePhase.SEMANTIC_ANALYSIS)
class BindLayoutASTTransformation extends AbstractASTTransformation {

    @Override
    void visit(ASTNode[] nodes, SourceUnit source) {

    }

}
