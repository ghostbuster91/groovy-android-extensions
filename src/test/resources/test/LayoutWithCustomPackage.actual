import pl.ghostbuster.grooid.*


@BindLayout({R.layout.layout_with_custom_package})
class LayoutWithCustomPackageTestCase {


}
