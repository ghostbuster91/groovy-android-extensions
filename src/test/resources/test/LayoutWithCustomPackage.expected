import android.widget.LinearLayout
import android.view.View
import pl.ghostbuster.grooid.*
import com.custom.packagename.*

class LayoutWithCustomPackageTestCase {

    public final LinearLayout rootLayout
    public final CustomObject nestedImageView


    public LayoutWithCustomPackageTestCase(View itemView){
          rootLayout = (LinearLayout) itemView.findViewById(R.id.root_layout)
          nestedImageView = (CustomObject) itemView.findViewById(R.id.nested_image_view)
    }
}
